packages/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -490,6 +497,23 @@ async function resolveDependency (
   ctx: ResolutionContext,
   options: ResolveDependencyOptions,
 ): Promise<PkgAddress | null> {
+  if (options.currentDepth === 0 && !options.update) {
+    const linkedDep = ctx.directLinkedPackages.find((directLinkedPackage) => directLinkedPackage.alias === wantedDependency.alias)
```

Comment:
this should be done in a separate function that is only called for the direct deps